opentelemetry-api/tests/context/base_context.py

```diff
@@ -0,0 +1,77 @@
+# Copyright 2020, OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+from logging import ERROR
+
+from opentelemetry import context
+
+
+def do_work() -> None:
+    context.attach(context.set_value("say", "bar"))
+
+
+class ContextTestCases:
```

Comment:
Is this wrapper class necessary? I was under the impression the idea was for the context implementation test cases just to inherit from the class declared here, for example:

```python
from unittest import TestCase, main


class ParentTests(TestCase):

    def test_0(self):
        assert True

    def test_1(self):
        assert False


class ChildTests(ParentTests):

    def test_2(self):
        assert True

    def test_3(self):
        assert False


if __name__ == "__main__":
    main()
```
```
.F.F.F
======================================================================
FAIL: test_1 (__main__.ChildTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "testin_parent.py", line 10, in test_1
    assert False
AssertionError

======================================================================
FAIL: test_3 (__main__.ChildTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "testin_parent.py", line 19, in test_3
    assert False
AssertionError

======================================================================
FAIL: test_1 (__main__.ParentTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "testin_parent.py", line 10, in test_1
    assert False
AssertionError

----------------------------------------------------------------------
Ran 6 tests in 0.001s

FAILED (failures=3)
```
Six test cases are run, that means that the child test class includes all of the parent class test cases too.

Reply:
So I created the class because my expectation here is that there would only be 4 tests:
```
test_0
test_1
test_2
test_3
```
Instead not wrapping the class does
```
test_0
test_0
test_0

Reply:
Ok, I see. Maybe this?:

```python
from unittest import TestCase, main


class ParentTests:

    def test_0(self):
        assert True

    def test_1(self):
        assert False


class Child0Tests(ParentTests, TestCase):

    def test_2(self):
        assert True

    def test_3(self):
        assert False


class Child1Tests(ParentTests, TestCase):

    def test_2(self):
        assert True

    def test_3(self):
        assert False


if __name__ == "__main__":
    main()
```
```
.F.F.F.F
======================================================================
FAIL: test_1 (__main__.Child0Tests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "testin_parent.py", line 10, in test_1
    assert False
AssertionError

======================================================================
FAIL: test_3 (__main__.Child0Tests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "testin_parent.py", line 19, in test_3
    assert False
AssertionError

======================================================================
FAIL: test_1 (__main__.Child1Tests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "testin_parent.py", line 10, in test_1
    assert False
AssertionError

======================================================================
FAIL: test_3 (__main__.Child1Tests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "testin_parent.py", line 28, in test_3
    assert False
AssertionError

----------------------------------------------------------------------
Ran 8 tests in 0.001s

FAILED (failures=4)
```

Reply:
yeah, this would work, but it would require changing all the assertions in the base test class to not use unittest.TestCase defined ones, otherwise pylint gets mad.

Reply:
Seems like this is a common enough problem for `pylint` to have a special [something](https://docs.pylint.org/en/1.6.0/faq.html) to deal with it:

> 5.2 How do I avoid “access to undefined member” messages in my mixin classes?
>To do so you have to set the ignore-mixin-members option to “yes” (this is the default value) and to name your mixin class with a name which ends with “mixin” (whatever case).

Reply:
This then requires updates to the base class to either ignore setUp/tearDown casing or update the method names. At this point, I'd rather just go with the wrapper class unless others feel strongly towards mixins

Reply:
:smile: no worries, let's keep it as it is :+1: